code/modules/mob/living/carbon/xenomorph/castes/Warrior.dm

```diff
@@ -101,6 +111,11 @@
 			L.pulledby = src
 			visible_message(SPAN_XENOWARNING("\The [src] grabs [L] by the throat!"), \
 			SPAN_XENOWARNING("You grab [L] by the throat!"))
+			lunging = TRUE
+			addtimer(CALLBACK(src, .proc/stop_lunging), get_xeno_stun_duration(L, 2) SECONDS + SShuman.next_fire - world.time)
```

Comment:
This easily on the codebase's Most Wanted bugs/hacks list and it shouldn't be reused
It caused issues before (polio) and it still is (all stuns last 1.5 to 2.5x longer than tehy should due to stacking)

Thanksfully you shouldn't need it for this case, because you'll get the precise duration off the timer, without the need for tick compensation